### PR TITLE
Make some ZAS lists lazy

### PR DESCRIPTION
**Pull request by Lohikar**
This PR converts some lists in ZAS to be only created when needed instead of always existing and pointlessly using memory. Might lead to reduced server memory usage.

### CODE REVIEW

File: code/ZAS/ConnectionGroup.dm [hunk 1]
```diff
@@ -215,7 +215,7 @@ Class Procs:
 	. = ..()
 
 /connection_edge/unsimulated/erase()
-	A.edges.Remove(src)
+	LAZYADD(A.edges, src)
```
Comment (skull132) [inline code/ZAS/ConnectionGroup.dm]: `.Remove` => `LAZYADD()` ???

_Doing things at 2AM is the best answer, obviously._

File: code/ZAS/Fire.dm [hunk 2]
```diff
@@ -52,21 +53,21 @@ turf/proc/hotspot_expose(exposed_temperature, exposed_volume, soh = 0)
...
+				LAZYREMOVE(fire_tiles, T)
+				LAZYREMOVE(fuel_objs, fuel)
 	else
 		for(var/turf/simulated/T in fire_tiles)
 			if(istype(T.fire))
 				T.fire.RemoveFire()
 			T.fire = null
-		fire_tiles.Cut()
-		fuel_objs.Cut()
+		LAZYCLEARLIST(fire_tiles)
```
Comment (skull132) [inline code/ZAS/Fire.dm]: Shouldn't these two be followed up with `UNSETEMPTY`? Unless `.Cut()` already sets them to `null`, but I doubt this.

File: code/ZAS/Zone.dm [hunk 3]
```diff
@@ -154,11 +155,14 @@ Class Procs:
...
 
+	LAZYINITLIST(graphic_add)
+	LAZYINITLIST(graphic_remove)
 	if(air.check_tile_graphic(graphic_add, graphic_remove))
 		for(var/turf/simulated/T in contents)
 			T.update_graphic(graphic_add, graphic_remove)
-		graphic_add.len = 0
-		graphic_remove.len = 0
+
+		LAZYCLEARLIST(graphic_add)
```
Comment (skull132) [inline code/ZAS/Zone.dm]: Again, `UNSETEMPTY()`?

File: code/ZAS/ConnectionGroup.dm (hunk 1 above)
Comment (Lohikar) [inline code/ZAS/ConnectionGroup.dm]: Whoops.

File: code/ZAS/Fire.dm (hunk 2 above)
Comment (Lohikar) [inline code/ZAS/Fire.dm]: I was just leaving the lists hanging around as it's fairly likely they'll be used again after they've been created.

Comment (skull132) [inline code/ZAS/Fire.dm]: Doesn't that like, negate the whole purpose of using lazy lists, though..?

Comment (Lohikar) [inline code/ZAS/Fire.dm]: This list should only be created during fires anyways.

**Comment by Lohikar** [comment]
Requested changes made.

**Pull request merged**